src/main/scala/com/typesafe/conductr/sbt/ConductR.scala

```diff
@@ -30,18 +34,39 @@ private[conductr] object ConductR {
   val actorSystemAttrKey = AttributeKey[ActorSystem]("sbt-conductr-actor-system")
 
   def loadBundle(apiVersion: String, bundle: URI, config: Option[URI], loadTimeout: Timeout, state: State): String = {
-    def get[A](key: SettingKey[A]) =
-      getOpt(key)
-        .fold(Bad(One(s"Setting ${key.key.label} must be defined!")): A Or One[String])(Good(_))
-    def getOpt[A](key: SettingKey[A]) =
-      Project.extract(state).getOpt(key)
-    def doLoadBundle(normalizedName: String, version: String, system: String, roles: Set[String], nrOfCpus: Double, memory: Bytes, diskSpace: Bytes) = {
+    def getSchedulingParams = {
+      val baseBundleConf = parseBundleConf(bundle)
+      val overlayBundleConf = Accumulation.convertOptionToCombinable(config.map(parseBundleConf(_))).combined
+      Accumulation.withGood(baseBundleConf, overlayBundleConf)(mergeSchedulingParams)
+    }
+    def parseBundleConf(bundle: URI) =
+      try {
+        val zipFile = new ZipFile(bundle.getPath)
+        val zipEntry = zipFile.entries.asScala.find(_.getName contains "bundle.conf")
+        val config = zipEntry.map(e => ConfigFactory.parseReader(new InputStreamReader(zipFile.getInputStream(e))))
+        Or.from(config, One(s"Zip file $bundle doesn't contain a bundle.conf file."))
+      } catch {
+        case _: IOException  => Bad(One(s"File $bundle doesn't exist."))
+        case _: ZipException => Bad(One(s"File $bundle has not a valid zip format."))
+      }
+    def mergeSchedulingParams(baseConf: Config, overlayConfOpt: Option[Config]) =
+      (
+        mergeConfigKey[String](baseConf, overlayConfOpt, _.getString("name")),
+        mergeConfigKey[String](baseConf, overlayConfOpt, _.getString("compatibilityVersion")),
+        mergeConfigKey[String](baseConf, overlayConfOpt, _.getString("systemVersion")),
+        mergeConfigKey[String](baseConf, overlayConfOpt, _.getString("system")),
+        mergeConfigKey[Set[String]](baseConf, overlayConfOpt, _.getStringList("roles").asScala.toSet),
+        mergeConfigKey[Double](baseConf, overlayConfOpt, _.getDouble("nrOfCpus")),
+        mergeConfigKey[Long](baseConf, overlayConfOpt, _.getLong("memory")),
+        mergeConfigKey[Long](baseConf, overlayConfOpt, _.getLong("diskSpace")))
+    def mergeConfigKey[A](baseConf: Config, overlayConfOpt: Option[Config], f: Config => A): A =
+      overlayConfOpt match {
+        case None              => f(baseConf)
+        case Some(overlayConf) => Try(f(overlayConf)).getOrElse(f(baseConf))
+      }
```

Comment:
Why have all this merge logic when typesafe config can merge for us? HOCON can be merged in many different ways - more sophisticated that what we have here.

Other than that, the approach looks reasonable.